code/game/objects/structures/flora.dm

```diff
@@ -94,13 +95,25 @@
 	death()
 	return ..()
 
-/obj/structure/flora/pottedplant/attackby(obj/item/weapon/W, mob/user)
-	if (W.edge)
-		user.visible_message(span("warning", "[user] cuts down the [src]"))
-		death()
-		return 1
+/obj/structure/flora/pottedplant/attackby(obj/item/W, mob/user)
+	if(do_after(user, 20, act_target = src))
+		if(!stored_item)
+			stored_item = W
+			to_chat(user,"<span class='notice'>You hide [W] in [src]</span>")
+			return
+		else
+			to_chat(user,"<span class='notice'>There is something hidden in [src]</span>")
+			return
 	return ..()
 
+/obj/structure/flora/pottedplant/attack_hand(mob/user)
+	if(do_after(user, 40, act_target = src))
+		if(!stored_item)
+			user.visible_message("<span class='notice'>Searches [src] for hidden items</span>","<span class='notice'>There is nothing hidden in [src]</span>")

```

Comment:
I feel like the `for hidden items` is superfluous here. This also won't make sense visually.

Also periods.

Reply:
Maybe something like: `user.visible_message("<span class='notice'>[user] searches [src].</span>", /* etc */)`.

Maybe the search message should be displayed before do_after.